src/main/webapp/script.js

```diff
@@ -11,3 +11,88 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/*
+ * Displays navbar authentication buttons according to login status.
+ */
+function fetchLogin() {
+  fetch('/login').then(response => response.json()).then(user => {
+    // If user is logged in, show logout button in navbar.
+    if (user.loggedIn) {
+      // If logged in user has no nickname, redirect to nickname setup page.
+      if(!user.isUserRegistered) {
+        window.location.replace(user.authenticationUrl);
+      }
+
+      // Delete signup button.
+      const signupButtonNavbar = document.getElementById('signup');
+      signupButtonNavbar.innerHTML = '';
+
+      // Create logout button.
+      const logoutButton = document.createElement('button');
+      logoutButton.setAttribute('type', 'button');
+      const logoutUrl = 'window.location.href = \"' + user.authenticationUrl + '\"';
+      logoutButton.setAttribute('onclick', logoutUrl);
+      logoutButton.classList.add('btn');
+      logoutButton.classList.add('btn-outline-success');
+      logoutButton.innerHTML = 'Log Out';
+
+      // Create navbar item to hold logout button.
+      const logoutButtonItem = document.createElement('li');
+      logoutButtonItem.classList.add('nav-item');
+      logoutButtonItem.appendChild(logoutButton);
+    
+      // Append logout button to navbar.
+      const logoutButtonNavbar = document.getElementById('login');
+      logoutButtonNavbar.innerHTML = '';
+      logoutButtonNavbar.appendChild(logoutButtonItem);
+    // If user is logged out, show signup and login buttons in navbar.
+    } else {
+      // Create signup button.
```

Comment:
same as above
```suggestion
    } else {
      // If user is logged out, show signup and login buttons in navbar.
      // Create signup button.
```

Reply:
> But these comments describe the whole if block, so I think they should precede the if statement. Otherwise, it seems I am describing the variables below, don't you think?

Reply:
Done!